### PR TITLE
Support more output formats from kubeclarity-cli analyze

### PR DESCRIPTION
**Pull request by sambetts-cisco**
This PR contains commits required to support more output formats for kubeclarity-cli analyze.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -266,6 +266,32 @@ registry:
...
+| CycloneDX JSON (default) | cyclonedx-json |
+| CycloneDX XML | cyclonedx-xml |
+| SPDX JSON | spdx-json |
+| SPDX Tag Value | spdx-tv |
+| Syft JSON | syft-json |
+
+
+> ***WARNING***  
+> kubeclarity processes CycloneDX internally, the other formats are supported
+> through a conversion so may not support all the same fields/information.
```
Comment (akpsgit) [inline README.md:286]: Maybe:
... through a conversion. The conversion process can be lossy, so not all fields/information are promised to be present in the result output.

File: cli/cmd/convert.go [hunk 2]
```diff
@@ -64,21 +61,30 @@ func convertSBOM(cmd *cobra.Command, _ []string) {
...
-		return true
+func ConvertSBOMFile(inputSBOMFile string, outputSBOMFile string, outputFormat converter.SbomFormat) error {
+	cdxBom, err := converter.GetCycloneDXSBOMFromFile(inputSBOMFile)
+	if err != nil {
+		return fmt.Errorf("failed to get CycloneDX SBOM from file: %v", err)
+	}
+
+	bomBytes, err := converter.CycloneDxToBytes(cdxBom, outputFormat)
+	if err != nil {
+		return fmt.Errorf("failed to convert CycloneDX to syft format: %v", err)
```
Comment (akpsgit) [inline cli/cmd/convert.go:82]: syft format?

File: cli/cmd/convert.go [hunk 3]
```diff
@@ -64,21 +61,30 @@ func convertSBOM(cmd *cobra.Command, _ []string) {
...
+	}
+
+	if err := utils.WriteSBOM(bomBytes, outputSBOMFile); err != nil {
+		return fmt.Errorf("failed to write syft SBOM to file %s: %v", outputSBOMFile, err)
```
Comment (akpsgit) [inline cli/cmd/convert.go:86]: same

File: shared/pkg/converter/cyclonedx.go [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+		return CycloneDxToNativeFormatBytes(sbom, format)
+	case SpdxJSON, SpdxTV, SyftJSON:
+		return CycloneDxToNonNativeFormatBytes(sbom, format)
+	case Unknown:
+	default:
+	}
+	return nil, fmt.Errorf("can not convert cyclonedx SBOM to unknown format")
+}
+
+func CycloneDxToNativeFormatBytes(sbom *cdx.BOM, format SbomFormat) ([]byte, error) {
```
[the remark this answers is not in the document]
Comment (sambetts-cisco) [reply shared/pkg/converter/cyclonedx.go:75]: renaming these functions, your comment made me realise that they aren't good names

**Review by akpsgit — DISMISSED** [review]

**Pull request merged**